[ADD] l10n_es_ticketbai_api: send invoices to the Basque Country tax …

ljsalvatierra-binovo opened the pull request:
…agencies (API).

Base addon for l10n_es_ticketbai.
Added Python library dependencies to requirements.txt.

There is another pull request for the main addon `l10n_es_ticketbai`.

oihane commented:
Fallos del PEP8
```
l10n_es_ticketbai_api/models/ticketbai_invoice_line.py:53:42: E126 continuation line over-indented for hanging indent
l10n_es_ticketbai_api/models/ticketbai_invoice_line.py:63:46: E126 continuation line over-indented for hanging indent
l10n_es_ticketbai_api/models/ticketbai_invoice.py:577:89: E501 line too long (89 > 88 characters)
```

Y los errores de los test igual están relacionados con esto que comentan https://github.com/OCA/l10n-spain/pull/1667#issuecomment-819379305

ljsalvatierra-binovo commented:
> Fallos del PEP8
> 
> ```
> l10n_es_ticketbai_api/models/ticketbai_invoice_line.py:53:42: E126 continuation line over-indented for hanging indent
> l10n_es_ticketbai_api/models/ticketbai_invoice_line.py:63:46: E126 continuation line over-indented for hanging indent
> l10n_es_ticketbai_api/models/ticketbai_invoice.py:577:89: E501 line too long (89 > 88 characters)
> ```
> 
> Y los errores de los test igual están relacionados con esto que comentan [#1667 (comment)](https://github.com/OCA/l10n-spain/pull/1667#issuecomment-819379305)

Gracias por el aviso Oihane, mi IDE me ha jugado una mala pasada. He vuelto a subir una nueva versión.

ljsalvatierra-binovo commented:
¿Alguna idea de cuál puede ser el problema con runbot?
![image](https://user-images.githubusercontent.com/23120306/114833963-651ba100-9dd0-11eb-9352-3cbb0c201098.png)

js-landoo commented:
@ljsalvatierra-binovo he hecho una prueba funcional y está correcto. He añadido un review de aprobación.
El error de runbot entiendo que es lo mismo que nos ha pasado a nosotros con la versión de python que decía Pedro, así que lo doy por bueno.
El resto, animaros a validar el módulo para que Luis pueda seguir con el siguiente. 
Las instrucciones para probarlo son las mismas del la version 12 de este mismo módulo. Lo teneis en el PR #1667 
Gracias.

js-landoo commented:
@ljsalvatierra-binovo por cierto, en la 12 hemos evitado el cr.commit del código replanteando un poco el flujo de envíos. Si quieres échale un vistazo porque el travis lo ha pasado pero sigue indicándolo como error porque en efecto, no es recomendable su uso.
Lo que hacemos es un try antes del envío y si falla salimos del while con un break y un return controlado. Con esto Odoo cierra bien la transacción sin necesidad de commit. Y las facturas que se hayan podido subir antes del error, quedan bien.

ljsalvatierra-binovo commented:
> @ljsalvatierra-binovo por cierto, en la 12 hemos evitado el cr.commit del código replanteando un poco el flujo de envíos. Si quieres échale un vistazo porque el travis lo ha pasado pero sigue indicándolo como error porque en efecto, no es recomendable su uso.
> Lo que hacemos es un try antes del envío y si falla salimos del while con un break y un return controlado. Con esto Odoo cierra bien la transacción sin necesidad de commit. Y las facturas que se hayan podido subir antes del error, quedan bien.

Buenas tardes @js-landoo, no veo dónde se hace el `try` antes del envío.
https://github.com/OCA/l10n-spain/blob/acd7bd6f5789a0ee8b40bbe0e68e9bd1dfc65e6b/l10n_es_ticketbai_api/models/ticketbai_invoice.py#L534:L587

Gracias por vuestro tiempo.

Bilbonet commented on l10n_es_ticketbai_api/README.rst:
```diff
@@ -0,0 +1,103 @@
+Para instalar esté módulo se necesita:
+
+#. Los módulos base y base_setup
+#. Las librerías Python OpenSSL, xmlsig, cryptography, qrcode y xmltodict.
```
Añadir la librería requests_pkcs12

ljsalvatierra-binovo replied:
Hola @Bilbonet, he subido una nueva versión añadiendo la librería, gracias por el aviso.

Bilbonet reviewed (approved):
Pruebas funcionales correctas según los comentarios en:
https://github.com/OCA/l10n-spain/pull/1667#issuecomment-820225800

ljsalvatierra-binovo commented:
He corregido el caso de factura no sujeta a IVA, estaba indicando el importe total, el cual siempre es cero, en vez de indicar la base imponible.

https://github.com/OCA/l10n-spain/blob/34546dcd970bf8f9fd30a4d3183674ce9e78bd5f/l10n_es_ticketbai_api/models/ticketbai_invoice.py#L898

ao-landoo commented:
> He corregido el caso de factura no sujeta a IVA, estaba indicando el importe total, el cual siempre es cero, en vez de indicar la base imponible.
> 
> https://github.com/OCA/l10n-spain/blob/34546dcd970bf8f9fd30a4d3183674ce9e78bd5f/l10n_es_ticketbai_api/models/ticketbai_invoice.py#L898

Nos apuntamos el cambio, gracias.

Un detalle, Travis os ha dado error por un tema de SII, nos sucedió lo mismo. Habría que ejecutar de nuevo Travis si no estoy equivocado.

Otra cosa @ljsalvatierra-binovo, en vez de hacer force-pushes, no es mejor realizar commits? Es mucho mas cómodo para ver que cambios se han hecho.

pedrobaeza commented:
Los force-push se pueden ver igualmente en GitHub, así que es la opción preferida para mantener limpio el historial de commits.

ljsalvatierra-binovo commented:
Aprovechando que tienen lo servidores de pruebas en mantenimiento y fuera de servicio, al hacer pruebas, hemos comprobado que no se guardaba correctamente la respuesta de la Hacienda.

https://github.com/OCA/l10n-spain/pull/1666/files#diff-fdc5e0d982782f94c2710e5320dacc50c0ab4f2f3d2f18ffec67ebd3a51f47b3R50-R68

ljsalvatierra-binovo commented:
Se ha modificado la vista de formulario `TicketBAI Instalaciones` para mostrar la licencia con caracteres ocultos, como una contraseña.

https://github.com/OCA/l10n-spain/pull/1666/files#diff-2272b5134901c3d23a5550e67ecffc732b4c69e0f361f08149cff92dfc774f1aR16

ljsalvatierra-binovo commented:
¿Se requiere la aprobación de alguien más para realizar el merge? Gracias.

enriquemartin reviewed (approved):
Pruebas funcionales satisfactorias! 
Una vez más, agradecer el currazo!

arfbin commented:
Se sacan los ficheros XMLs, QR y certificado p12 de la BD al filestore

pedrobaeza commented:
Travis está en rojo

ghost commented:
- Se añade la versión del software en la instalación de TicketBAI.
- Al habilitar TicketBAI en la compañia, se le asignara la instalación por defecto en caso de que solo haya una (que será lo mas común)

ljsalvatierra-binovo committed:
[ADD] l10n_es_ticketbai_api: send invoices to the Basque Country tax agencies (API).
Base addon for l10n_es_ticketbai.
Added Python library dependencies to requirements.txt.

ljsalvatierra-binovo commented:
Hola, ¿se puede hacer merge? Para hacer el PR del siguiente módulo de Ticketbai, gracias.

enriquemartin commented:
Buenas! ¿Se puede hacer merge de esto? Así se puede avanzar en el resto de módulos que dependen de este.
Gracias!

pedrobaeza commented:
/ocabot merge nobump
